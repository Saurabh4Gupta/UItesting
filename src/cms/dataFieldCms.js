export const dataFieldCms = {
  createNewDataRequest: 'Create new data request',
  editDataRequest: 'Edit data request',
  localMarketLabel: 'Local market',
  localMarketHint: 'Choose the market you wish to create this data request for',
  nameLabel: 'Data request name',
  backLabel:'Back to clients',
  nameHint:
    "Make sure it's clear to help others identify it.Max 255 characters",
  briefLabel: 'Briefing',
  briefHint:
    'Tell us in detail what this data request is for, max 4000 characters',
  templateFileLabel: 'Add tracker template file',
  templateFileHint:
    'Only .xls, .xlsx, .csv, .xlsb, .xlsm, .xlt, .xltm, .xltx files in 20mb or less',
  uploadFileLabel:
    'To upload a new file,drag and drop you file below or browse using the link',
  uploadFileLargeMessage: 'File must be less than 20MB',
  labelFileTypeNotAllowed: 'File must be of type .xls, .xlsx, .csv, .xlsb, .xlsm, .xlt, .xltm, .xltx',
  uploadButtonText:'Upload new file',
  actualDataLabel: 'Actual data',
  versionHistoryTitle:'Version history',
  actualDataHint: 'Number of months of actual productivity data',
  forecastDataLabel: 'Forecast data',
  forecastDataHint: 'Number of months of forecasted productivity data',
  dueDateLabel: 'Due date',
  dueDateHint: 'Enter the date that you need this data request by',
  assignToLabel: 'Assign to',
  assignToHint: `Only the person or team you assign this to will have access. Once you click 'Create' ,
  they will be sent email notifications asking them to fill in the data request.`,
  create: 'Create',
  upload: 'Upload',
  cancel: 'Cancel',
  productivityDatarequestHeading: 'Productivity data requests',
  productivityDatarequestCaption: ' The ongoing and complete productivity data requests created for this client',
  selectPlaceHolder: 'Please select',
  datePlaceHolder: 'Date',
  viewDetails: 'View Details',
  viewClientsHeading: 'My clients',
  viewClientsCaption: 'Choose the client you wish to view',
  completeLabel: 'Complete',
  ongoingLabel: 'Ongoing',
  emptyProductivityDatarequestCaption:
    'There are currently no active Productivity data requests',
  emptyLabel: 'Create new data request',
  edit: 'Edit',
  actual: 'Actuals',
  forecast: 'Forecast',
  overdue: 'Overdue',
  ongoing: 'Ongoing',
  complete: 'Complete',
  lastUpdate: 'Last Update',
  moveToComplete: 'Move to Complete',
  moveToOnGoing: 'Move to Ongoing',
  delete: 'Delete',
  deletePopConfimationMessage:'Once deleted, this data request will no longer be available.',
  deleteButtonCancelMessage:'No, Cancel',
  deleteButtonDeleteMessage:'Yes, Delete',
  popUpTitle:'Are you sure you want to delete?',
  emptyCompletedProductivityDatarequestCaption:
    'No Productivity data requests have been moved to this section yet.',
  editRequest: 'Edit request',
  save: 'Save',
  productivityRequest: 'Productivity request',
  showMore: 'Show more',
  reportingYear: 'Reporting year',
  productivityDataRequested: 'Productivity data requested',
  actualCapitalized: 'Actual',
  forecastCapitalized: 'Forecast',
  dueBy: 'Due by',
  assignedTo: 'Assigned to',
  trackerTemplate: 'Tracker template',
  close: 'Close',
  reportingYearLabel: 'Reporting year',
  reportingYearHint: 'The reporting year this data request is for',
  toastRequestCreated: 'Data request created',
  toastRequestEdited: 'Data request edited',
  toastRequestDeleted: 'Data request deleted',
  toastFileUploaded: 'File uploaded',
  toastMovedToOngoing: 'Data request moved to ongoing',
  toastMovedToComplete: 'Data request moved to complete',
  currentVersionText:'current version',
  loadMoreText:'Load more',
  uploadFileErrorMessage:'Upload tracker template',
  moveToCompleteConfirmation: 'Are you sure you want to move this to complete?',
  moveToCompleteDescription: 'You will not be abale to make any more edits to this request unless it is moved back to "Ongoing".',
  deletePopUpConfirmation: 'Are you sure you want to delete?',
  deletePopUpDescription: 'Once deleted, this data request will no longer be available.',
  noCancel: 'No, cancel',
  yesContinue: 'Yes, continue',
  yesDelete: 'Yes, delete',
}
