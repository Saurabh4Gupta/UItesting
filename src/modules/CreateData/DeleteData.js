import React from 'react';
import { Modal, Button } from '@dentsu-ui/components/dist/cjs/components/Modal';

const [modalOpen, setModalOpen] = useState(false);
const DeleteData = () => {
  const handleDelete = () => {
    console.log('Hello deleted');
    setModalOpen(false);
  }
  return (
    <>
      <Modal isOpen={modalOpen} onClose={() => setModalOpen(false)}>
        <Modal.Header hasCloseButton title="Delete this data requests" />
        <Modal.Body>
          <p>
            Are you sure want to delete this request?
          </p>
        </Modal.Body>
        <Modal.Footer>
          <Button variant="secondary" onClick={() => setModalOpen(false)}>
            cancel
          </Button>
          <Button onClick={handleDelete}>
            Send
          </Button>
        </Modal.Footer>
      </Modal>
    </>
  )
}
export default DeleteData;
