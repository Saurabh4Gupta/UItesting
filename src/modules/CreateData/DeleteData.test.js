import React from 'react';
import { shallow } from 'enzyme';
import DeleteData from './DeleteData';
import { dataFieldCms } from '../../cms';

describe('Success test cases for: DeleteData component', () => {
  let wrapper;
  const props = {
    setModalOpen: jest.fn(),
    deleteRequest: jest.fn(),
    deleteModalData: {},
    handleToggleData: jest.fn(),
    setIsMoveToCompleteModel: jest.fn(),
    moveToCompleteModelData: {},
    isDeleteModal: true,
    // moveToCompleteConfirmation: 'Are you sure you want to move this to complete?',
    // moveToCompleteDescription: 'You will not be able to make any more edits to this request unless it is moved back to "Ongoing".',
    // deletePopUpConfirmation: 'Are you sure you want to delete?',
    // deletePopUpDescription: 'Once deleted, this data request will no longer be available.',
    // noCancel: 'No, cancel',
    // yesContinue: 'Yes, continue',
    // yesDelete: 'Yes, delete',
  }

  describe('Snapshot test for  DeleteData component', () => {
    beforeEach(() => {
      wrapper = shallow(<DeleteData {...props} />);
    });

    it('Test to match the snapsot', async (done) => {
      expect(wrapper).toMatchSnapshot();
      done();
    });

    it('Test to match the snapsot', async () => {
      const fn = wrapper.find('ForwardRef').at(0).prop('onClose');
      expect(fn).toBeInstanceOf(Function);
      fn();
    });

    // it('Test to match the snapsot', () => {
    //   const fn1 = wrapper.find('ForwardRef').at(2).find('Button');
    //   console.debug(fn1)
    //   expect(fn1.prop('onClick')())

    // });
    it('Test click event', () => {
      wrapper.find({ children: dataFieldCms.noCancel }).simulate('click');
    });

    it('Test click event', () => {
      wrapper.find({ children: dataFieldCms.yesContinue  }).simulate('click');
    });

    it('Renders correctly', () => {
      expect(wrapper.exists()).toBe(true);
    });
  });
});

describe('Failure test cases for: Duplicate data page', () => { });
