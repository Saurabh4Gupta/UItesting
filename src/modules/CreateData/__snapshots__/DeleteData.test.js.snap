// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Success test cases for: DeleteData component Snapshot test for  DeleteData component Test to match the snapsot 1`] = `
<Fragment>
  <ForwardRef
    isFullHeight={false}
    onClose={[Function]}
    width="500px"
  >
    <ForwardRef
      hasCloseButton={true}
      title="Are you sure you want to move this to complete?"
    />
    <ForwardRef>
      <p>
        You will not be able to make any more edits to this request unless it is moved back to "Ongoing".
      </p>
    </ForwardRef>
    <ForwardRef>
      <ForwardRef
        onClick={[Function]}
        variant="secondary"
      >
        No, cancel
      </ForwardRef>
      <ForwardRef
        onClick={[Function]}
      >
        Yes, continue
      </ForwardRef>
    </ForwardRef>
  </ForwardRef>
</Fragment>
`;
